dali/kernels/signal/fft/cufft_helper.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_SIGNAL_FFT_CUFFT_HELPER_H_
+#define DALI_KERNELS_SIGNAL_FFT_CUFFT_HELPER_H_
+
+#include <cufft.h>
+#include <utility>
+#include <string>
+#include "dali/core/cuda_error.h"
+#include "dali/core/format.h"
+
+namespace dali {
+
+class CUFFTError : public std::runtime_error {
+ public:
+  explicit CUFFTError(cufftResult result, const char *details = nullptr)
+  : std::runtime_error(Message(result, details))
+  , result_(result) {}
+
+  static const char *ErrorString(cufftResult result) {
+    switch (result) {
+      case CUFFT_SUCCESS:
+        return "cuFFT operation was successful";
+      case CUFFT_INVALID_PLAN:
+        return "cuFFT was passed an invalid plan handle";
+      case CUFFT_ALLOC_FAILED:
+        return "cuFFT failed to allocate memory";
```

Comment:
Same for `CUFFT_ALLOC_FAILED`?

Reply:
Likewise, it's a public static function which may be used to translate the cuFFT error code in the absence of actually throwing the exception.